dts/bindings/power/nordic,nrf-power.yaml

```diff
@@ -13,3 +13,15 @@ properties:
 
   interrupts:
     required: true
+
+  dcdc-components:
+    type: boolean
+    description: |
+        Enable if DC/DC components are on the board which allows running of the SoC's DC/DC
+        instead of LDO.
+
+  hv-dcdc-components:
+    type: boolean
+    description: |
+        Enable if HV (high voltage) DC/DC components are on the board which allows running of
+        the SoC's DC/DC instead of LDO.
```

Comment:
I think we need to discuss this more in detail. By looking at the [specs](https://infocenter.nordicsemi.com/topic/ps_nrf52840/power.html?cp=5_0_0_4_2), it looks like it's time to create a simple regulator driver, described as a subnode of `nordic,nrf-power`. A deprecation path would be good as well.

Reply:
The problem is that this cannot be deprecated, the default options for boards as they are now is to have this enabled by default, which is the same as it has here. The problem arises because if someone has an overlay to disable the Kconfig, then the old Kconfig would indeed be disabled but would have no impact on the new Kconfigs. The power mode selection is performed as part of the startup in the system_nrf52840.c file so moving it to a driver seems like a much bigger job

Reply:
I think it looks messy that we now has a node where the the status property effectively means nothing, because the Kconfig enables and disables.

If switching dc/dc to devicetree, then we should use `status = "disabled";` / `status = "okay";` instead of current mix / mess.

Reply:
Have updated to use regulator binding and `regulator-boot-on` property